server/src/main/java/io/druid/discovery/DruidLeaderClient.java

```diff
@@ -221,14 +232,21 @@ public String findCurrentLeader()
     }
   }
 
-  private String getCurrentKnownLeader(final boolean cached)
+  private String getCurrentKnownLeader(final boolean cached) throws IOException
   {
-    return currentKnownLeader.accumulateAndGet(
+    final String leader = currentKnownLeader.accumulateAndGet(
         null,
         (current, given) -> current == null || !cached ? pickOneHost() : current
     );
+
+    if (leader == null) {
+      throw new IOE("No known leader");
```

Comment:
"No known server" . `pickOneHost()` is only supposed to find a server not necessarily leader... it is finding leader now only because of we're keeping ServerDiscoverySelector for backward compatibility. that would be removed in next release and then it would only rely upon the list of available overlord/coordinator servers in `internal-discovery`

Reply:
Ah, okay, changed it.